esp/bindings/http/platform/httpservice.cpp

```diff
@@ -847,6 +858,25 @@ int CEspHttpServer::onGetXslt(CHttpRequest* request, CHttpResponse* response, co
 }
 
 
+int CEspHttpServer::unsupported(const char* httpRequestMethod)
+{
+    // Per https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/, the value
+    // of http.request.method must be "known to the instrumentation" and is case sensitive.
+    if (strieq(httpRequestMethod, HEAD_METHOD) || strieq(httpRequestMethod, "PUT") || strieq(httpRequestMethod, "DELETE") || strieq(httpRequestMethod, "CONNECT") || strieq(httpRequestMethod, "TRACE") || strieq(httpRequestMethod, "PATCH"))
```

Comment:
I'm not crazy about the worst case overhead (non-standard method name would result in multiple unneeded string compares) and the best case scenario is that we report the http.method in an unsupported request type in the expected form...
I'm leaning towards reporting httpRequestMethod.toUpper() even if we potentially violate otel's standards.